src/runtime/components/elements/Ladder.vue

```diff
@@ -0,0 +1,185 @@
+<template>
+  <ol :class="ui.base" v-bind="attrs">
+    <template v-for="(wrapper, key) in computedSteps" :key="key">
+      <li :class="ui.container">
+        <div :class="ui.step">
+          <div :class="ui.indicator">
+            <div v-if="hasSeparators" :class="ui.separator.wrapper">
+              <div :class="[ui.separator.base, ui.separator.color, key < 1 ? 'invisible' : '']" />
+              <div :class="[ui.separator.base, ui.separator.color, key > computedSteps.length - 2 ? 'invisible' : '']" class="ml-10" />
+            </div>
+            <div :class="ui.icon.wrapper">
+              <component
+                :is="wrapper.step.click || wrapper.step.to ? 'UButton' : 'div'"
+                :class="[ui.icon.base, ui.icon.rounded, ui.icon.shadow, ui.icon.size, ui.icon.ring, getIconClass(wrapper)]"
+                type="button"
+                variant="ghost"
+                v-bind="omit(wrapper.step, ['label', 'click', 'icon'])"
+                @click="wrapper.step.click"
+              >
+                <UIcon v-if="wrapper.step.icon" :name="wrapper.step.icon" />
+                <div v-else>
+                  {{ key + 1 }}
+                </div>
+              </component>
+            </div>
+          </div>
+
+          <div :class="[ui.label.base, ui.label.size, getLabelClass(wrapper)]">
+            {{ wrapper.step.label }}
+          </div>
+        </div>
+      </li>
+    </template>
+  </ol>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, toRef } from 'vue'
+import type { ComputedRef, PropType } from 'vue'
+import { twJoin } from 'tailwind-merge'
+import { useUI } from '../../composables/useUI'
+import { mergeConfig, omit } from '../../utils'
+import type { Strategy, LadderSize, LadderStep, LadderStepWrapper, LadderStepColor } from '../../types'
+// @ts-expect-error
+import appConfig from '#build/app.config'
+import { ladder } from '#ui/ui.config'
+import UButton from './Link.vue'
+
+const config = mergeConfig<typeof ladder>(appConfig.ui.strategy, appConfig.ui.ladder, ladder)
+
+export default defineComponent({
+  components: {
+    UButton
+  },
+  inheritAttrs: false,
+  props: {
+    modelValue: {
+      type: Number,
+      default: 0
+    },
+    steps: {
+      type: Array as PropType<String[] | LadderStep[]>,
+      default: () => []
+    },
+    separators: {
+      type: Boolean,
+      default: true
+    },
+    color: {
+      type: String as PropType<LadderStepColor>,
+      default: () => config.default.color,
+      validator (value: string) {
+        return [...appConfig.ui.colors, 'gray'].includes(value)
+      }
+    },
+    inactiveColor: {
+      type: String as PropType<LadderStepColor>,
+      default: () => config.default.inactiveColor,
+      validator (value: string) {
+        return [...appConfig.ui.colors, 'gray'].includes(value)
+      }
+    },
+    click: {
```

Comment:
Shouldn't this be handled in `items` directly and emit a `click` instead of using a prop?

Reply:
You know? You're right if you think that the normal implementation for this would be to not have any click behavior and let the changes be done outside, like a form.

I just thought that it would be faster to implement a way to let the dev handle the click for changing steps in one variable rather than adding the handler on each item. 

Can I set then the `change-on-click` prop to add a default click handler to all items to change to that step when clicked?